language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/traces/TraceRecord.java

```diff
@@ -16,29 +16,125 @@
 
 package org.ballerinalang.langserver.extensions.ballerina.traces;
 
+import com.google.gson.JsonObject;
+
+import java.util.UUID;
+
 /**
  * Model class for trace log.
  */
 public class TraceRecord {
-    private Trace meta;
-    private String record;
+    private Message message;
+    private String rawMessage;
+    private String id;
+    private String logDate;
+    private String millis;
+    private String sequence;
+    private String logger;
+    private String level;
+    private String sourceClass;
+    private String sourceMethod;
+    private String thread;
+
+    TraceRecord(Message message, JsonObject record, String rawMessage) {
+        this.message = message;
+        this.rawMessage = rawMessage;
+        this.id = UUID.randomUUID().toString();
+        this.logDate = record.get("logDate").getAsString();
+        this.millis = record.get("millis").getAsString();
+        this.sequence = record.get("sequence").getAsString();
+        this.logger = record.get("logger").getAsString();
+        this.level = record.get("level").getAsString();
+        this.sourceClass = record.get("sourceClass").getAsString();
+        this.sourceMethod = record.get("sourceMethod").getAsString();
+        this.thread = record.get("thread").getAsString();
+    }
+
+    public String getRawMessage() {
+        return rawMessage;
+    }
+
+    public String getLogger() {
+        return logger;
+    }
+
+    public String getSequence() {
+        return sequence;
+    }
+
+    public String getThread() {
+        return thread;
+    }
+
+    public String getMillis() {
+        return millis;
+    }
+
+    public String getLogDate() {
+        return logDate;
+    }
+
+    public String getSourceMethod() {
+        return sourceMethod;
+    }
+
+    public String getSourceClass() {
+        return sourceClass;
+    }
 
-    TraceRecord(Trace meta, String record) {
-        this.setMeta(meta);
-        this.setRecord(record);
+    public String getLevel() {
+        return level;
     }
 
-    public void setMeta(Trace meta) {
-        this.meta = meta;
+    public Message getMessage() {
+        return message;
     }
-    public Trace getMeta() {
-        return meta;
+
+    public String getId() {
+        return id;
+    }
+
+    public void setRawMessage(String rawMessage) {
```

Comment:
Do we need these setter methods? Considering `TraceRecord` as immutable we can omit setter methods for this class. Then it is easier to write test cases.